presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -283,6 +284,11 @@ public SystemSessionProperties(
                         featuresConfig.isLegacyOrderBy(),
                         false),
                 booleanSessionProperty(
+                        SUPPORT_NULL_ON_LHS_OF_SEMI_JOIN,
```

Comment:
> I didn't measure that.
Please do, because memory might the difference that is more visible than performance difference.

> I think we use this knowledge when handling joins.

If that would be the case then there would be a bug in join handling because of projections.

> I think both kinds of information are valuable:

Sure they are, but they have to be separated into two kind of extractors.

> Perhaps in JOIN both information kinds are useful and in SEMI JOIN only the latter (effective predicate) is?

Still if you get predicate `x=3 or x is null` you don't want to give up on PP completely, but you can PP predicate like:
`z=3 or pass_first_row()` or similar